### PR TITLE
EWC-60 Healthy / unhealthy status (stuck in restart), registry fix

### PR DESCRIPTION
**Pull request by elukashick**

### CODE REVIEW

File: daemon/src/org/eclipse/iofog/element/Registry.java [hunk 1]
```diff
@@ -72,18 +76,23 @@ public String getUserEmail() {
...
-				Objects.equals(userEmail, registry.userEmail);
+
+		if (isPublic != registry.isPublic) return false;
+		return url.equals(registry.url);
 	}
 
 	@Override
 	public int hashCode() {
-		return Objects.hash(url, userEmail);
+		int result = url.hashCode();
```
Comment (ekrylovich) [inline daemon/src/org/eclipse/iofog/element/Registry.java]: Use Objects hash method

File: daemon/src/org/eclipse/iofog/field_agent/FieldAgent.java [hunk 2]
```diff
@@ -385,16 +385,19 @@ private void loadRegistries(boolean fromFile) {
...
-						.setUserEmail(registry.getString("useremail"))
-						.build();
-				registries.add(result);
+						.setIsPublic(registry.getBoolean("ispublic"));
+				if (!registry.getBoolean("ispublic")) {
+					registryBuilder.setUserName(registry.getString("username"))
+							.setPassword(registry.getString("password"))
+							.setUserEmail(registry.getString("useremail"));
+					//unused fields
+					//.setSecure(registry.getBoolean("secure"))
```
Comment (ekrylovich) [inline daemon/src/org/eclipse/iofog/field_agent/FieldAgent.java]: remove

File: daemon/src/org/eclipse/iofog/process_manager/RestartStuckChecker.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		static final RestartStuckChecker hcc = new RestartStuckChecker();
+	}
+
+	public static RestartStuckChecker getInstance() {
+		return SingletonHolder.hcc;
+	}
+
+
+	public boolean isStuck(String elementId) {
+		List<LocalDateTime> datesOfRestart = restarts.computeIfAbsent(elementId, k -> new ArrayList<>());
```
Comment (ekrylovich) [inline daemon/src/org/eclipse/iofog/process_manager/RestartStuckChecker.java]: rename "k"

File: daemon/src/org/eclipse/iofog/process_manager/RestartStuckChecker.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	private static final int ABNORMAL_NUMBER_OF_RESTARTS = 5;
+
+	private RestartStuckChecker() {
+	}
+
+	public static class SingletonHolder {
+		static final RestartStuckChecker hcc = new RestartStuckChecker();
+	}
+
+	public static RestartStuckChecker getInstance() {
```
Comment (ekrylovich) [inline daemon/src/org/eclipse/iofog/process_manager/RestartStuckChecker.java]: remove singleton and useless synchronization

File: daemon/src/org/eclipse/iofog/field_agent/FieldAgent.java [hunk 5]
```diff
@@ -385,16 +385,15 @@ private void loadRegistries(boolean fromFile) {
...
-						.setSecure(registry.getBoolean("secure"))
-						.setCertificate(registry.getString("certificate"))
-						.setRequiresCertificate(registry.getBoolean("requirescert"))
-						.setUserName(registry.getString("username"))
-						.setPassword(registry.getString("password"))
-						.setUserEmail(registry.getString("useremail"))
-						.build();
-				registries.add(result);
+						.setIsPublic(registry.getBoolean("ispublic"));
+				if (!registry.getBoolean("ispublic")) {
```
Comment (epankou) [inline daemon/src/org/eclipse/iofog/field_agent/FieldAgent.java]: maybe put this "public" logic into corresponding setters? what do you think

**Pull request merged**